### PR TITLE
Deprecate InstrumentationLibraryInfo

### PR DESCRIPTION
**Pull request by jack-berg**
Related to #4190.

- Adjust ReadableSpan, SpanData to include accessor to InstrumentationScopeInfo, deprecate #getInstrumentationLibraryInfo()
- Adjust MetricData to include accessor to InstrumentationScopeInfo, deprecate #getInstrumentationLibraryInfo()
- Adjust LogData to include accessor to InstrumentationScopeInfo, deprecate #getInstrumentationLibraryInfo()

### CODE REVIEW

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/logs/InstrumentationScopeLogsMarshaler.java [hunk 1]
```diff
@@ -9,42 +9,42 @@
...
   private static int calculateSize(
-      InstrumentationLibraryMarshaler instrumentationLibrary,
+      InstrumentationScopeMarshaller instrumentationScope,
       byte[] schemaUrlUtf8,
       List<Marshaler> logMarshalers) {
     int size = 0;
     size +=
         MarshalerUtil.sizeMessage(
-            InstrumentationLibraryLogs.INSTRUMENTATION_LIBRARY, instrumentationLibrary);
+            InstrumentationLibraryLogs.INSTRUMENTATION_LIBRARY, instrumentationScope);
```
Comment (anuraaga) [inline exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/logs/InstrumentationScopeLogsMarshaler.java:47]: Just merged the proto update, so think you need to merge main and update the keys now when marshaling

File: sdk/common/src/test/java/io/opentelemetry/sdk/common/InstrumentationLibraryInfoTest.java [hunk 2]
```diff
@@ -10,18 +10,21 @@
 
 import org.junit.jupiter.api.Test;
 
-/** Tests for {@link InstrumentationLibraryInfo}. */
+/** Tests for {@link io.opentelemetry.sdk.common.InstrumentationLibraryInfo}. */
```
Comment (anuraaga) [inline sdk/common/src/test/java/io/opentelemetry/sdk/common/InstrumentationLibraryInfoTest.java:13]: Looks like some qualified imports got automatically added by intellij or something

Comment (jack-berg) [reply sdk/common/src/test/java/io/opentelemetry/sdk/common/InstrumentationLibraryInfoTest.java:13]: You have to qualify the imports, even in comments, or else you get an error for using deprecated classes.

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/logs/InstrumentationScopeLogsMarshaler.java (hunk 1 above)
Comment (jack-berg) [reply exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/logs/InstrumentationScopeLogsMarshaler.java:47]: Actually, it looks like the proto PR to change the field names hasn't merged yet: https://github.com/open-telemetry/opentelemetry-proto/pull/362

File: sdk/common/src/test/java/io/opentelemetry/sdk/common/InstrumentationLibraryInfoTest.java (hunk 2 above)
Comment (anuraaga) [reply sdk/common/src/test/java/io/opentelemetry/sdk/common/InstrumentationLibraryInfoTest.java:13]: Ah thought if it's in the same package that wouldn't be an issue but maybe not

**Review by jkwatson — APPROVED** [review]

**Comment by jkwatson-verta** [comment]
We all missed that this was a breaking change, since we added methods to public interfaces without a default implementation. :(

**Pull request merged**